### PR TITLE
Install azcopy when not using an auto pool

### PR DESCRIPTION
**Pull request by jordeu**
- Download as a job resource `azcopy` for user provider Azure Batch pools.
- Add the boolean property `azure.batch.installAzCopy` to enable/disable this behaviour
- By default `azcopy` is only downloaded when the pool is not created by nextflow.

### CODE REVIEW

File: plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy [hunk 1]
```diff
@@ -339,6 +339,13 @@ class AzBatchService implements Closeable {
 
         final resFiles = new ArrayList(10)
 
+        if( config.batch().installAzCopy!=Boolean.FALSE && !config.batch().canCreatePool() ) {
+            log.debug "[AZURE BATCH] installing azcopy as task resource"
+            resFiles << new ResourceFile()
+                    .withHttpUrl('https://nf-xpack.s3-eu-west-1.amazonaws.com/azcopy/linux_amd64_10.8.0/azcopy')
```
Comment (abhi18av) [inline plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:345]: Hi @jordeu ,

I was wondering whether there's a way we could make sure that we're always using the latest release?

Comment (pditommaso) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:345]: The opposite, silent updates break your code.

Comment (jordeu) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:345]: I also think that is better to update this more manually. I am watching the releases [here](https://github.com/Azure/azure-storage-azcopy/releases) to get notified when there is a new release.

**Comment by pditommaso** [comment]
Ok, can you replace the link to aws s3 with `https://github.com/nextflow-io/azcopy-tool/raw/linux_amd64_10.8.0/azcopy`

File: plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy [hunk 2]
```diff
@@ -342,7 +342,7 @@ class AzBatchService implements Closeable {
         if( config.batch().installAzCopy!=Boolean.FALSE && !config.batch().canCreatePool() ) {
             log.debug "[AZURE BATCH] installing azcopy as task resource"
             resFiles << new ResourceFile()
-                    .withHttpUrl('https://nf-xpack.s3-eu-west-1.amazonaws.com/azcopy/linux_amd64_10.8.0/azcopy')
+                    .withHttpUrl('https://github.com/nextflow-io/azcopy-tool/raw/linux_amd64_10.8.0/azcopy')
                     .withFilePath('.nextflow-bin/azcopy')
```
Comment (pditommaso) [inline plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:346]: It think there's a smilar one in the pool creation start task

Comment (jordeu) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:346]: Oops, you're right.

Comment (jordeu) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:346]: done

File: plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy [hunk 3]
```diff
@@ -532,7 +539,7 @@ class AzBatchService implements Closeable {
         def resourceFiles = new ArrayList(10)
         
         resourceFiles << new ResourceFile()
-                .withHttpUrl("https://nf-xpack.s3-eu-west-1.amazonaws.com/azcopy/linux_amd64_10.8.0/azcopy")
+                .withHttpUrl("https://github.com/nextflow-io/azcopy-tool/raw/linux_amd64_10.8.0/azcopy")
                 .withFilePath('azcopy')
 
         def poolStartTask = new StartTask()
```
Comment (pditommaso) [inline plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:545]: OK, let's use a constant, easy to forget otherwise.

**Comment by pditommaso** [comment]
Made some refactor and merged manually 9e9337d3

**Pull request closed without merging**